source/aws_profile.c

```diff
@@ -198,6 +203,7 @@ struct profile_file_parse_context {
     int parse_error;
     int current_line_number;
     bool has_seen_profile;
+    bool is_sso_session;
```

Comment:
there will be other section types, so you might want to consider basing the logic on whether its profile section or not, rather then whether its not session token section (looks like profile requires separate logic from other section types).

Reply:
Thanks, fixed by section_type enum.